build.sh

```diff
@@ -5,80 +5,109 @@
 # 2022-07-31: adding extra rc.local, setting rc build script
 # 2022-08-01: improvements for passwordless root, git version build script, accessip not in use
 # 2022-08-02: switch to using git submodule for mfsbsd, drop git clone step for that repo
+# 2022-08-04: improvements to script, clearing shellcheck errors
 
 # this script must be run as root
 if [ "$EUID" -ne 0 ]; then
     echo "Please run this script as root user"
     exit
 fi
 
+# 'set -eo pipefail' means we need to be careful with grep and zero search results
+# because grep exits with an error code, causing the pipeline to fail, script exit
+#
+# a workaround is to enclose grep as follows, where e is search term:
+#
+# cmd | { grep e || :; } | cmd
+#
+set -eo pipefail
+
+exit_error() {
+    echo "$*" 1>&2
+    exit 1;
+}
+
+usage() {
+    cat <<-EOF
+	Usage: $(basename "${BASH_SOURCE[0]}") [-hbu] [-k /path/to/authorized_keys] 
+	
+	-h Show help
+	-b Build without uploading
+	-u Build with upload to remote host
+	-k /path/to/authorized_keys (can safely ignore, another opportunity to copy in SSH keys on image boot!)
+	
```

Comment:
I think this tab character can go away (so no trailing whitespace) and it should still format correctly (probably best to test once).